wrangler-core/src/main/java/io/cdap/directives/column/SetType.java

```diff
@@ -22,44 +22,81 @@
 import io.cdap.wrangler.api.Arguments;
 import io.cdap.wrangler.api.Directive;
 import io.cdap.wrangler.api.DirectiveExecutionException;
+import io.cdap.wrangler.api.DirectiveParseException;
 import io.cdap.wrangler.api.ExecutorContext;
+import io.cdap.wrangler.api.Optional;
 import io.cdap.wrangler.api.Row;
 import io.cdap.wrangler.api.annotations.Categories;
 import io.cdap.wrangler.api.lineage.Lineage;
 import io.cdap.wrangler.api.lineage.Mutation;
+import io.cdap.wrangler.api.parser.Bool;
 import io.cdap.wrangler.api.parser.ColumnName;
 import io.cdap.wrangler.api.parser.Identifier;
+import io.cdap.wrangler.api.parser.Numeric;
+import io.cdap.wrangler.api.parser.Text;
 import io.cdap.wrangler.api.parser.TokenType;
 import io.cdap.wrangler.api.parser.UsageDefinition;
 import io.cdap.wrangler.utils.ColumnConverter;
 
+import java.math.RoundingMode;
 import java.util.List;
 
 /**
- * A Wrangler step for converting data type of a column
+ * A Wrangler step for converting data type of column
  * Accepted types are: int, short, long, double, float, string, boolean and bytes
+ * When decimal type is selected, can also specify the scale and whether to round or not
  */
 @Plugin(type = "directives")
 @Name(SetType.NAME)
 @Categories(categories = {"column"})
-@Description("Converting data type of a column.")
+@Description("Converting data type of a column. Optional arguments scale, " +
+  "round and rounding-mode are used only when type is decimal.")
 public final class SetType implements Directive, Lineage {
   public static final String NAME = "set-type";
+
+  // Default scale that is assumed in CDAP output schema
+  public static final int DEFAULT_SCALE = 3;
```

Comment:
By default we should do exactly as we did before. We can't change the existing logic

Reply:
This is the default scale used when rounding is enabled but scale is not specified. The existing logic is not changed.

Reply:
I believe now it just convert to BigDecimal. E.g. currently if you have `1.23456` value it will remain `1.23456` value and would work if you set schema properly. With this change it's going to fail as default scale is 3 and  RoundingMode.UNNECESSARY.

Reply:
Changed the code to preserve current logic.